pkg/cmd/roachtest/roachtestutil/disconnect.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package roachtestutil
+
+import (
+	"context"
+
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test"
+	"github.com/stretchr/testify/require"
+)
+
+// Disconnect takes a set of nodes and disconnects each node from all others in
+// the list. This function requires root privileges.
+//
+// TODO (msbutler):  consider a different api: Disconnect(t test.Test, c
+// cluster.Cluster, ctx context.Context, node int, toDisconnect []int) where we
+// disconnect a node from all nodes in toDisconnect. Adding iptable rules on
+// both sides shouldn't be necessary.
+func Disconnect(ctx context.Context, t test.Test, c cluster.Cluster, nodes []int) {
+	if c.IsLocal() {
+		t.Status("skipping iptables disconnect on local cluster")
```

Comment:
Should we panic? This considerably changes the behavior of the test, and is easily overlooked.

Reply:
good point. fixed.